gossip/src/cluster_info.rs

```diff
@@ -1088,34 +1094,23 @@ impl ClusterInfo {
                 .min() // Boot the oldest evicted vote by wallclock.
                 .map(|(_ /*wallclock*/, ix)| ix)
         };
-        vote_index.unwrap_or(num_crds_votes)
+        if num_crds_votes < MAX_LOCKOUT_HISTORY as u8 {
+            // Do not evict if there is space in crds
+            num_crds_votes
```

Comment:
theoretically this might actually override existing votes.
say there only 3 votes in crds table but with indices 0, 1, 3.

it is the same with the old code though.